services/localfile/client/utils.go

```diff
@@ -577,3 +578,11 @@ func ListRemote(ctx context.Context, conn *proxy.Conn, listRequest ListRequest)
 	}
 	return ret, nil
 }
+
+const fileModeSizeInBits = 12
+
+func ParseFileMode(modeStr string) (uint16, error) {
```

Comment:
This was my desperate attempt to avoid adding more code to `client.go`. It already has over 1.4k LOC.
I agree that the function doesn't have to be public and since I have not been able to find any obvious way of creating package-private symbols in Go I followed your suggestion and moved `ParseFileMode` to `client.go`